hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -1219,14 +1243,13 @@ private Pair<HoodieWriteConfig, Schema> getHoodieClientConfigAndWriterSchema(Sch
     return Pair.of(config, returnSchema);
   }
 
-  private Schema getSchemaForWriteConfig(Schema targetSchema) {
+  private Schema getSchemaForWriteConfig(Schema targetSchema, HoodieTableMetaClient metaClient) {
```

Comment:
Would it make sense to make `metaClient` a member variable of `StreamSync` instead of passing it around?

Reply:
I've gone back and forth on this myself. I think that the original intention is that the meta client always reflects the current state and therefore doesn't need to exist outside the scope of a single sync. I am curious to hear what others think and I am open to changing this.

Reply:
I think it's better keep the meta client within the scope of a single sync and pass it around for reuse without keeping a `metaClient` member which is error-prone.